packages/cw-orch-core/src/environment/state.rs

```diff
@@ -77,3 +77,33 @@ impl<S: StateInterface> StateInterface for Rc<RefCell<S>> {
         (**self).borrow().deploy_details()
     }
 }
+
+impl<S: StateInterface> StateInterface for Rc<S> {
+    fn get_address(&self, contract_id: &str) -> Result<Addr, CwEnvError> {
+        (**self).get_address(contract_id)
+    }
+
+    fn set_address(&mut self, contract_id: &str, address: &Addr) {
+        (*Rc::make_mut(self)).set_address(contract_id, address)
```

Comment:
Just making a note here that `make_mut` will clone the inner value and set the address through that.

Reply:
Only if we create another rc to that state, right?